enterprise/app/api_keys/api_keys.tsx

```diff
@@ -576,3 +573,73 @@ function newFormState<T extends ApiKeyFields>(request: T): FormState<T> {
     request,
   };
 }
+
+interface ApiKeyFieldProps {
+  value: string;
+}
+
+interface ApiKeyFieldState {
+  value: string;
+  isCopied: boolean;
+  hideValue: boolean;
+  displayValue: string;
+}
+
+const ApiKeyFieldDefaultState: ApiKeyFieldState = {
+  value: "",
+  isCopied: false,
+  hideValue: true,
+  displayValue: "************",
+};
+
+class ApiKeyField extends React.Component<ApiKeyFieldProps, ApiKeyFieldState> {
+  state = ApiKeyFieldDefaultState;
+
+  componentDidMount() {
+    this.setState({ value: this.props.value });
+  }
+
+  // onClick handler function for the copy button
+  private handleCopyClick() {
+    const copyTextToClipboard = async (text: string) => {
+      if ("clipboard" in navigator) {
+        return await navigator.clipboard.writeText(text);
+      } else {
+        return document.execCommand("copy", true, text);
+      }
+    };
+
+    // Asynchronously call copyTextToClipboard
+    copyTextToClipboard(this.state.value).then(() => {
+      // If successful, update the isCopied state value
+      this.setState({ isCopied: true });
+      setTimeout(() => {
+        this.setState({ isCopied: false });
+      }, 1500);
+    });
+  }
+
+  // onClick handler function for the hide/reveal button
+  private toggleHideValue() {
+    this.setState({
+      hideValue: !this.state.hideValue,
+      displayValue: this.state.hideValue ? this.state.value : ApiKeyFieldDefaultState.displayValue,
+    });
+  }
+
+  render() {
+    const { isCopied, hideValue, displayValue } = this.state;
+
+    return (
+      <div className="api-key-value">
+        <button className="api-key-value-copy" onClick={this.handleCopyClick.bind(this)}>
+          {isCopied ? <Check style={{ stroke: "green" }} className="icon" /> : <Copy className="icon" />}
+        </button>
+        <button className="api-key-value-hide" onClick={this.toggleHideValue.bind(this)}>
+          {hideValue ? <Eye className="icon" /> : <EyeOff className="icon" />}
+        </button>
+        <span>{displayValue}</span>
```

Comment:
nit: I think it is more conventional to have the value on the left and the controls on the right.

Otherwise looks great :)

Reply:
No strong opinion here. I will move it to the right to see how it look